src/agent/onefuzz-agent/src/tasks/task_logger.rs

```diff
@@ -0,0 +1,604 @@
+            }))
+        }
+    }
+
+    #[tokio::test]
+    async fn test_task_logger_normal_messages() -> Result<()> {
+        let events = Arc::new(RwLock::new(HashMap::new()));
+        let log_writer = Box::new(TestLogWriter {
+            id: 0,
+            events: events.clone(),
+            max_size: 1,
+        });
+
+        let blob_logger = TaskLogger {
+            job_id: Uuid::new_v4(),
+            task_id: Uuid::new_v4(),
+            machine_id: Uuid::new_v4(),
+            logging_interval: Duration::from_secs(1),
+            log_buffer_size: 1,
+            polling_interval: Duration::from_secs(1),
+        };
+
+        let (tx, rx) = tokio::sync::broadcast::channel(16);
+        tx.send(LogEvent::Trace((log::Level::Info, "test1".into())))?;
+        tx.send(LogEvent::Trace((log::Level::Info, "test2".into())))?;
+        tx.send(LogEvent::Trace((log::Level::Info, "test3".into())))?;
+        tx.send(LogEvent::Trace((log::Level::Info, "test4".into())))?;
+        tx.send(LogEvent::Trace((log::Level::Info, "test5".into())))?;
+
+        let _res =
+            tokio::time::timeout(Duration::from_secs(5), blob_logger._start(rx, log_writer)).await;
+
+        let x = events.read().unwrap();
+
+        for (k, values) in x.iter() {
+            println!("{}", k);
+            for v in values {
+                println!(" {:?}", v);
+            }
+        }
+
+        assert_eq!(x.keys().len(), 5, "expected 5 groups of messages");
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_task_logger_big_messages() -> Result<()> {
+        let events = Arc::new(RwLock::new(HashMap::new()));
+        let log_writer = Box::new(TestLogWriter {
+            id: 0,
+            events: events.clone(),
+            max_size: 2,
+        });
+
+        let blob_logger = TaskLogger {
+            job_id: Uuid::new_v4(),
+            task_id: Uuid::new_v4(),
+            machine_id: Uuid::new_v4(),
+            logging_interval: Duration::from_secs(3),
+            log_buffer_size: 2,
+            polling_interval: Duration::from_secs(1),
+        };
+
+        let (tx, rx) = tokio::sync::broadcast::channel(16);
+        tx.send(LogEvent::Trace((log::Level::Info, "test1".into())))?;
+        tx.send(LogEvent::Trace((log::Level::Info, "test2".into())))?;
+        tx.send(LogEvent::Trace((log::Level::Info, "test3".into())))?;
+        tx.send(LogEvent::Trace((log::Level::Info, "test4".into())))?;
+        tx.send(LogEvent::Trace((log::Level::Info, "test5".into())))?;
+
+        let _res =
+            tokio::time::timeout(Duration::from_secs(15), blob_logger._start(rx, log_writer)).await;
+
+        let x = events.read().unwrap();
+
+        for (k, values) in x.iter() {
+            println!("{}", k);
+            for v in values {
+                println!(" {:?}", v);
+            }
+        }
+
+        assert_eq!(x.keys().len(), 3, "expected 3 groups of messages");
+        Ok(())
+    }
+
+    #[tokio::test]
+    #[ignore]
```

Comment:
Are these tests marked with `#[ignore]` intended to be run manually by specifying the env var?

Reply:
correct.